### ISSUE TITLE
``unspecified-encoding`` when using ``Path().read_text()`` without encoding keyword argument

### ISSUE THREAD

**Issue by GeZZu** [PyCQA/pylint#5029]
### Bug description

```python
I will get unspecified-encoding even when I provide the encoding to Path().read_text() funciton.
For example Path().read_text("UTF-8").

I'll get rid of the warning by giving keyword argument Path().read_text(encoding="UTF-8")
```

### Command used

```shell
pylint test.py -e unspecified-encoding
```

### Pylint output

```shell
W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```

### Expected behavior

No warnings are given when providing encoding without keyword.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
```

**Comment by DanielNoord** [comment]
Based on the code in `pathlib` I don't think you are actually supplying an encoding, see:
```python
    def write_text(self, data, encoding=None, errors=None):
        """
        Open the file in text mode, write to it, and close the file.
        """
        if not isinstance(data, str):
            raise TypeError('data must be str, not %s' %
                            data.__class__.__name__)
        with self.open(mode='w', encoding=encoding, errors=errors) as f:
            return f.write(data)
```
If I read the code correctly, you are passing `"UTF-8"` as `data` instead of `encoding`.

**Comment by Pierre-Sassoulas** [comment]
I think it's for read_text and not write_text:
```python
    def read_text(self, encoding=None, errors=None):
        """
        Open the file in text mode, read it, and close the file.
        """
        with self.open(mode='r', encoding=encoding, errors=errors) as f:
            return f.read()
```

**Comment by DanielNoord** [comment]
My bad! Working on a fix now!
